designs/PackageReference-Extern-Alias.md

```diff
@@ -0,0 +1,171 @@
+```
+
+There is no need for a change here. 
+
+The value provided in the Aliases attribute will be pass through as far as NuGet is concerned. NuGet will not validate that the value provided is a valid alias, that will be done at build time. 
+
+The implementation of this feature spans multiple components. 
+Specifically the work items as follows: 
+
+* NuGet.Client makes the changes for processing the new metadata and adding to the project.assets.json
+* The build tasks on (.NET Core SDK side)[https://github.com/dotnet/sdk/blob/master/src/Tasks/Microsoft.NET.Build.Tasks/ResolvePackageAssets.cs] consume these changes
+* The build tasks on the (non-SDK based PackageReference)[https://github.com/dotnet/NuGet.BuildTasks/blob/master/src/Microsoft.NuGet.Build.Tasks/ResolveNuGetPackageAssets.cs] consume these changes.
+* Nomination updates on (project-system)[https://github.com/dotnet/project-system/blob/master/src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Rules/Dependencies/PackageReference.xaml] side.
+
+## Future Work
+
+* Package metadata for transitive package references will be considered in the future. For now the recommendation is to elevate that PackageReference to a direct dependency. 
+
+## Open Questions
+
+* Will create follow ups for the partner issues once this design has been reviewed.
+* In which version will this functionality be ready?
+
+## Considerations
+
+In addition to the proposed approach, 2 of other solutions were considered. 
+
+### Do nothing - Recommend the customers use the custom target workaround
+
+The workaround that allows per assembly granularity is the following:
+
+```xml
+  <Target Name="AddCustomAliases" BeforeTargets="FindReferenceAssembliesForReferences;ResolveReferences">
+    <ItemGroup>
+      <ReferencePath Condition="'%(FileName)' == '$AssembleFileName$' AND '%(ReferencePath.NuGetPackageId)' == '$PackageId$'">
+        <Aliases>$Alias$</Aliases>
+      </ReferencePath>
+    </ItemGroup>
+  </Target>
+```
+
+Pros:
+
+* A very powerful solution, that covers both direct and transitive dependencies.
+* Allows per file granularity, if multiple assemblies are brought in by a package, each can have(or not) their own alias.
+
+Cons:
+
+* Important enough feature that deserves an out of the box solution.
+* The workaround recommends taking a dependency on tasks and targets that are best not depended on by customers. Goes against the policy of minimizing the size of the project files.
+* Not obvious, unnecessarily exposes the customers to restore/build internals.  
+
+### Support `Aliases` as a PackageReference property & provide a way to address transitively dependency out of the box
+
+A future evolution of the current approach could be the addition of metadata for transitive packages. 
+
+```xml
+<ItemGroup>
+    <PackageReference Include="StackExchange.Redis.StrongName.Signed" Version="1.26.0" Aliases="signed" />
+</ItemGroup>
+```

```

Comment:
what's the difference between this one and the example above?? I think I'm missing something.

Reply:
I think this example needs to be removed and then we just have the transitive metadata proposal.

Reply:
Fixed it now.